app/src/main/java/com/duckduckgo/app/survey/ui/SurveyViewModel.kt

```diff
@@ -52,12 +57,21 @@ class SurveyViewModel @Inject constructor(
 
     val command: SingleLiveEvent<Command> = SingleLiveEvent()
     private lateinit var survey: Survey
+    private lateinit var source: SurveySource
+    private lateinit var lastActiveDay: String
     private var didError = false
 
-    fun start(survey: Survey) {
+    fun start(survey: Survey, source: SurveySource) {
         val url = survey.url ?: return
         this.survey = survey
-        command.value = Command.LoadSurvey(addSurveyParameters(url))
+        this.source = source
+        viewModelScope.launch {
+            lastActiveDay = when (source) {
+                SurveySource.IN_APP -> appDaysUsedRepository.getLastActiveDay()
+                SurveySource.PUSH -> appDaysUsedRepository.getPreviousActiveDay() ?: appDaysUsedRepository.getLastActiveDay()
```

Comment:
how about `getPreviousActiveDay` returns  `getLastActiveDay` if null, so this logic in not in the ViewModel?

Reply:
I wouldn't like to add this logic in the Repository because getting lastDay instead, is specific to this scenario. In other scenarios (if any) we would like to return null so we know there isn't a previous date. Does it make sense?

Reply:
no strong opinions, sounds good!